### DISCUSSION TITLE
Enable network visibility, WebRTC

### DISCUSSION THREAD

**Discussion by iulko** [celzero/rethink-app#907] [Q&A]
Hi, great job with the app.
Whats the difference between "Enable network visibility" and "Use all available networks".
Will "use all available networks" fix problems with apps like whatsapp etc not reciving audio calls?
Will integration with wireguard possibly help with that or its impossible to fix because of WebRTC and we will always need to bypass audio apps? 
I saw its the case in other firewalls for Android that this apps also dont work properly.

**Comment by ignoramous** [comment]
Add WhatsApp to "Bypass Universal" and audio calls should work. Or "Exclude" WhatsApp if you so prefer.

The "Enable network visibility" setting (in theory) should help with conference Video calls, but apps may or may not be making use of it via appropriate APIs.

**Comment by iulko** [comment]
Ok, I understand it can't work with any firewall in the app so the minimum is "Bypass Universal"

Don't want to create another discussion thread so three questions:

1. "Proxy" options are not available in lockdown mode, will wireguard support also not be available in this mode or is it different because rethink is the one that makes the connection?
2. Will wireguard not work with "Bypass Universal" apps or there should be no problem?
3. Is there a reason "Do not route priavate IPs" dont work in lockdown mode? Does android not let VPN access LAN IP in lockdown mode?

Sorry for opening so many issues and discussion but I just made a big review of all the options and this app is so great, there is just small things that are missing.

**Reply by ignoramous** [reply]
Don't worry about it. I like helping folks out! (:

1. Yes, WireGuard will be available in the *VPN Lockdown* mode, too.
2. Shouldn't be a problem, I don't think.
3. Yeah, *VPN Lockdown* means no packets (from apps) are sent outside the VPN tunnel. So, if the tunnel does not route Private IPs, outgoing packets to those IPs then have no where to go.

**Comment by iulko** [comment]
Hey, so audio calls don't seem to work with only "Bypass", they seem to need full exclude, so no VPN lockdown.
Is it the same for you?
Maybe its a conflict with "Block when source app is unknown" or "block port80"  but bypass should make it so this ones are not used?

**Reply by ignoramous** [reply]
WhatsApp Audio calls should ideally work (this is the first I have heard that they don't). Use Cloudflare or Google or Quad9 DNS and disable all rules and blocks and check if Audio calls then work? Likely some DNS rule is getting in the way.
